### ISSUE TITLE
declaration-colon-newline-after takes a colon inside a between-comment for the declaration's and writes its newline into the comment's text

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#499]
### What minimal example or steps are needed to reproduce the bug?

```css
a { b:
/*x:y*/ red; }
```

— a block comment standing between a declaration's colon and its value, spelling a colon of its own.

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/declaration-colon-newline-after": "always"
	}
}
```

### What did you expect to happen?

The declaration's colon has its newline: nothing to report. The colon inside the comment is the comment's text, not the declaration's.

### What actually happened?

The rule warns at the comment's colon — `Expected newline after ":"` at line 2 — and `--fix` writes the newline into the comment's text:

```css
a { b:
/*x:
y*/ red; }
```

### Notes

The walk of `declaration-colon-newline-after` over `raws.between` takes every colon it meets for the declaration's, comments included — on the example above the real colon passes silently and the comment's own is the one warned about, while `sharedRunsOf` and `declarationColonSpaceChecker` both find the declaration's colon as the first one standing outside a comment (`styleSearch`). Found by the review of the branch of [#355](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/355), whose `betweenTailAfterColon` had the same misreading before it landed; the pairs corpus spells no comment with a colon, so no oracle sees it. Measured on either side of that branch alike.
